File: solutions/csharp/leap/Leap.cs
public static class Leap {
    public static bool IsLeapYear(int year) =>
        year.HasFactorOf(4) && (!year.HasFactorOf(100) || year.HasFactorOf(400));
}

public static class IntExtensions {
    public static bool HasFactorOf(this int number, int divisor) =>
        number % divisor == 0;
}

File: solutions/typescript/leap/leap.ts
export function isLeap(year: number) : boolean {
  return isEvenlyDivisible(year, 4) && (!isEvenlyDivisible(year, 100) || isEvenlyDivisible(year, 400));
}

function isEvenlyDivisible(n1 : number, n2 : number) : boolean {
  return n1 % n2 == 0;
}